module/dkg/client.go

```diff
@@ -196,6 +318,22 @@ func (c *Client) SubmitParametersAndResult(indexMap flow.DKGIndexMap, groupPubli
 	ctx, cancel := context.WithTimeout(context.Background(), epochs.TransactionSubmissionTimeout)
 	defer cancel()
 
+	// TODO(mainnet27, #6792): remove this code block
+	{
+		v2Compatible, err := c.checkDKGContractV2Compatible(ctx)
+		// CASE 1: we know FlowDKG is not v2 compatible - fallback to v1 submission
+		if !v2Compatible && err == nil {
+			c.Log.Debug().Msg("detected FlowDKG version incompatible with protocol v2 - falling back to v1 result submission")
+			return c.submitResult_ProtocolV1(groupPublicKey, publicKeys)
+		}
+		// CASE 2: we aren't sure whether FlowDKG is v2 compatible - fallback to v1 submission
+		if err != nil {
+			c.Log.Warn().Err(err).Msg("unable to determine FlowDKG compatibility - falling back to v1 result submission")
+			return c.submitResult_ProtocolV1(groupPublicKey, publicKeys)
+		}
```

Comment:
This is probably not a question that we can address as part of this PR. When reading these lines, I am wondering what would happen if the protocol was on version 2 already, but script execution failed and we would submit with version 1 ... not sure how unlikely that is and what the consequences could be. Probably something best to be discussed in person.

Reply:
In that case, the v1 result submission would fail, and we would retry (execute both the script execution and resubmission again). So if script execution transiently fails, it's not a big deal -- if it fails consistently (where the transaction submission otherwise would have succeeded), it could cause a failure.